pkg/assisted/agentserviceconfig_test.go

```diff
@@ -0,0 +1,810 @@
+		{
+			exists: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+		)
+
+		if testCase.exists {
+			runtimeObjects = append(runtimeObjects, generateAgentServiceConfig())
+		}
+
+		testBuilder := buildTestASCBuilderWithFakeObjects(runtimeObjects)
+
+		aci, err := testBuilder.Get()
+		if testCase.exists {
+			assert.Nil(t, err)
+			assert.NotNil(t, aci)
+		} else {
+			assert.NotNil(t, err)
+			assert.Nil(t, aci)
+		}
+	}
+}
+func TestAgentServiceConfigCreate(t *testing.T) {
+	testCases := []struct {
+		exists bool
+	}{
+		{
+			exists: true,
+		},
+		{
+			exists: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+		)
+
+		if testCase.exists {
+			runtimeObjects = append(runtimeObjects, generateAgentServiceConfig())
+		}
+
+		testBuilder := buildTestASCBuilderWithFakeObjects(runtimeObjects)
+
+		result, err := testBuilder.Create()
+		assert.Nil(t, err)
+		assert.NotNil(t, result)
+		assert.Equal(t, agentServiceConfigName, result.Definition.Name)
+	}
+}
+func TestAgentServiceConfigUpdate(t *testing.T) {
+	testCases := []struct {
+		exists        bool
+		expectedError error
+	}{
+		{
+			exists:        true,
+			expectedError: nil,
+		},
+		{
+			exists:        false,
+			expectedError: errors.New("cannot update non-existent agentserviceconfig"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+		)
+
+		if testCase.exists {
+			runtimeObjects = append(runtimeObjects, generateAgentServiceConfig())
+		}
+
+		testBuilder := buildTestASCBuilderWithFakeObjects(runtimeObjects)
+
+		testBuilder.Definition.Spec.IPXEHTTPRoute = "enabled"
+
+		aci, err := testBuilder.Update(true)
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, aci.Object.Spec.IPXEHTTPRoute, "enabled")
+		}
+	}
+}
+func TestAgentServiceConfigDelete(t *testing.T) {
+	testCases := []struct {
+		exists        bool
+		expectedError error
+	}{
+		{
+			exists:        true,
+			expectedError: nil,
+		},
+		{
+			exists:        false,
+			expectedError: errors.New("agentserviceconfig cannot be deleted because it does not exist"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+		)
+
+		if testCase.exists {
+			runtimeObjects = append(runtimeObjects, generateAgentServiceConfig())
+		}
+
+		testBuilder := buildTestASCBuilderWithFakeObjects(runtimeObjects)
+
+		err := testBuilder.Delete()
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, testBuilder.Object)
+		}
+	}
+}
+func TestAgentServiceConfigDeleteAndWait(t *testing.T) {
+	testCases := []struct {
+		exists        bool
+		expectedError error
+	}{
+		{
+			exists:        true,
+			expectedError: nil,
+		},
+		{
+			exists:        false,
+			expectedError: errors.New("agentserviceconfig cannot be deleted because it does not exist"),
```

Comment:
Please change it. We have to return nil and set Object to nil if actual object doesn't exist.